istioctl/pkg/writer/pilot/testdata/singleStatus.txt

```diff
@@ -1,2 +1,2 @@
-NAME       CDS       LDS        EDS       RDS        ISTIOD      VERSION
-proxy2     STALE     SYNCED     STALE     SYNCED     istiod2     1.1
+NAME       CDS       LDS        EDS       RDS        NDS          ECDS         ISTIOD      VERSION
+proxy2     STALE     SYNCED     STALE     SYNCED     NOT SENT     NOT SENT     istiod2     1.1
```

Comment:
NDS and ECDS are not sent in majority of cases, I have concerns we will get a bunch of bugs that think something is broken even though its not

Reply:
As mentioned in previous comments, it is not shown by default. Status works the same way as it is for others - that is comparing nonce sent and acked. If there is a bug and if it is something that we can trace back to XDS implementation for that particular type then we might have to fix it to adhere to XDS protocol specification. 

I tested the case where sidecar DNS resolution is not enabled. In that case, status shown is `NOT SENT`.

Reply:
What are we trying to do with NDS and ECDS sync data?  There may be better ways to accomplish this...

Reply:
Extending proxy-status to show other XDS resource sync statuses as well. This PR does not change the output of `istioctl proxy-status` - we still show the same things by default - CDS,LDS,EDS,RDS. But that is available. We needed these additional sync statuses for reporting purposes. So I enhanced `statusgen`. Enhancing `istioctl proxy-status` is a next step. 

> There may be better ways to accomplish this...

Could you please suggest the same?

Reply:
@therealmitchconnors - The mechanism to determine status is the same. It is generic for various xDS types in Istiod.